core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1090,13 +1036,13 @@ final class Stream[+F[_], +O] private[fs2] (private val underlying: Pull[F, O, U
     * Not as powerful as `observe` since not all pipes can be represented by `O => F[_]`, but much faster.
     * Alias for `evalMap(o => f(o).as(o))`.
     */
-  def evalTap[F2[x] >: F[x]: Functor](f: O => F2[_]): Stream[F2, O] =
```

Comment:
Without this change, Dotty complains:

```scala
[error] 1040 |    evalMap(o => f(o).as(o))
[error]      |                 ^^^^^^^
[error]      |value as is not a member of F2[?]
[error]      |
[error]      |where:    F2 is a type in method evalTap with bounds >: F and <: [x] =>> Any
```

Rewriting as `evalMap(o => Functor[F2].as(f(o), o))` results in:

```scala
[error] 1040 |    evalMap(o => Functor[F2].as(f(o), o))
[error]      |                                ^^^^
[error]      |Found:    F2[?]
[error]      |Required: F2[Any]
[error]      |
[error]      |where:    F2 is a type in method evalTap with bounds >: F and <: [x] =>> Any
```